`ToggleGroupControl`: remove animated backdrop

KevinBatdorf opened the pull request:
<!-- Learn the overall process and best practices for pull requests at https://github.com/WordPress/gutenberg/blob/HEAD/docs/contributors/repository-management.md#pull-requests. -->

<!-- Gutenberg's license is in the process of updating to be dual-licensed under the GPL and MPL. As part of that transition, all new contributions are dual-licensed. For more information, see: https://github.com/WordPress/gutenberg/blob/trunk/LICENSE.md -->

## Description
<!-- Please describe what you have changed or added -->
This PR removes the backdrop animation component and replaces it with a simple background color change

This PR is an alternative solution to #37507 

Fixes #37506

## How has this been tested?
<!-- Please describe in detail how you tested your changes. -->
<!-- Include details of your testing environment, tests ran to see how -->
<!-- your change affects other areas of the code, etc. -->
Tested by adding a component with a background color and swapping between gradient and solid. Also, when the container is focused, pressing the left and right arrow keys properly applies the background color.

## Screenshots <!-- if applicable -->

https://user-images.githubusercontent.com/1478421/149639228-9f3b2caf-50bf-49a4-b6f7-ab531aacdaa1.mov

## Types of changes
<!-- What types of changes does your code introduce?  -->
<!-- Bug fix (non-breaking change which fixes an issue) -->
<!-- New feature (non-breaking change which adds functionality) -->
<!-- Breaking change (fix or feature that would cause existing functionality to not work as expected) -->

Bug fix

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style. <!-- Check code: `npm run lint`, Guidelines: https://developer.wordpress.org/coding-standards/wordpress-coding-standards/javascript/ -->
- [x] My code follows the accessibility standards. <!-- Guidelines: https://developer.wordpress.org/coding-standards/wordpress-coding-standards/accessibility/ -->
- [x] I've tested my changes with keyboard and screen readers. <!-- Instructions: https://github.com/WordPress/gutenberg/blob/HEAD/docs/contributors/accessibility-testing.md -->
- [x] My code has proper inline documentation. <!-- Guidelines: https://developer.wordpress.org/coding-standards/inline-documentation-standards/javascript/ -->
- [x] I've included developer documentation if appropriate. <!-- Handbook: https://developer.wordpress.org/block-editor/ -->
- [ ] I've updated all React Native files affected by any refactorings/renamings in this PR (please manually search all `*.native.js` files for terms that need renaming or removal). <!-- React Native mobile Gutenberg guidelines: https://github.com/WordPress/gutenberg/blob/HEAD/docs/contributors/code/native-mobile.md -->
- [ ] I've updated related schemas if appropriate. <!-- Reference: https://github.com/WordPress/gutenberg/tree/trunk/schemas -->

KevinBatdorf committed:
Remove backdrop feature

ciampo commented on packages/components/src/toggle-group-control/toggle-group-control/component.tsx:
```diff
@@ -49,7 +48,7 @@ function ToggleGroupControl(
 	const cx = useCx();
 	const containerRef = useRef();
-	const [ resizeListener, sizes ] = useResizeAware();
+	const [ resizeListener ] = useResizeAware();
```
It looks like `react-resize-aware` was only needed to resize the backdrop, and so I think we can remove this line and the import at the top of this file (cc'ing @ntsekouras for a confirmation!)

KevinBatdorf replied:
That makes sense actually. The listener isn't very useful if the `sizes` aren't being used. I removed it for now

ntsekouras replied:
Yeah, it seems it can be safely removed.

ciampo reviewed (commented):
Thank you @KevinBatdorf for opening this PR!

The code changes test well, and overall LGTM. Would you be able to add an entry to the [CHANGELOG](https://github.com/WordPress/gutenberg/blob/trunk/packages/components/CHANGELOG.md)?

The e2e failures don't seem related, maybe rebasing could be enough to have them pass.

KevinBatdorf committed:
Merge branch 'trunk' into fix/change-toggle-control-active-state-css

KevinBatdorf committed:
Remove resize listener

KevinBatdorf committed:
Remove resize listener

KevinBatdorf committed:
Add PR link to changelog

KevinBatdorf commented:
Hey @ciampo I added the changelog and removed the listener. I also had to update the test to remove the listener too. Some of the tests are still failing, but I don't think they are related to this change?

ntsekouras reviewed (approved):
Thanks for your work and all the iterations @KevinBatdorf! 👏 

LGTM and I agree with @ciampo about this being  a better short-term solution for the issue.
With a green CI let's 🚢

ciampo reviewed (approved):
Thank you Kevin, LGTM 🚀

jasmussen commented:
Is there anything we can do to follow up and get the animated backdrop back in the next few days? Although I appreciate the need to fix separate bugs, this is a significant design regression.

ciampo commented:
> Is there anything we can do to follow up and get the animated backdrop back in the next few days? Although I appreciate the need to fix separate bugs, this is a significant design regression.

Implementing this transition effectively and in a bug-free manner can prove quite complicated (see the complexity of the code changes in the last bug fix attempt in #37507), and I can't guarantee that I'll be able to come up with a solution in just a few days.

If, for the time being, you'd rather keep the older animation and reintroduce the bug that this PR fixed, I can go ahead an revert this PR. I will then work separately on a new implementation for the animation, and merge it when it's ready.

jasmussen commented:
We'll want to fix the bug, of course, but I'd prefer keeping the bug, and the backdrop until we can get both working at the same time. Thanks all for the work!
